### DISCUSSION TITLE
Outlook 365 v16.81 on Sonoma not syncing to google cal

### DISCUSSION THREAD

**Discussion by JohnGainor1** [derekantrican/GAS-ICS-Sync#406] [General]
Hi Derek,

Been using Cal Sync for years - good stuff.  Latest Outlook not syncing.

1. uninstalled old 'Cal Sync' script
2. copied and installed new 'GAS-ICS-Sync' script
3. entered ICS string
4. installed script and run - says execution complete (no errors)

no joy.  any ideas?  any issues with this outlook version?

Thanks!
John

<img width="1590" alt="Screenshot_1_22_24__9_54 AM" src="https://github.com/derekantrican/GAS-ICS-Sync/assets/157310179/79a2c4d6-3d35-4497-ac00-060f18f92f04">

**Comment by Lonestarjeepin** [comment]
What do you see in the execution log if you manually run startSync?  What you're showing is a successful install, but that function doesn't do anything other than create triggers really (i.e. it doesn't process calendar events).

**Comment by JohnGainor1** [comment]
Thanks for the swift response.

It fails with startSync
Any ideas?

***@***.***

From: Lonestarjeepin ***@***.***>
Reply-To: derekantrican/GAS-ICS-Sync ***@***.***>
Date: Monday, January 22, 2024 at 10:03 AM
To: derekantrican/GAS-ICS-Sync ***@***.***>
Cc: John Gainor ***@***.***>, Author ***@***.***>
Subject: Re: [derekantrican/GAS-ICS-Sync] Outlook 365 v16.81 on Sonoma not syncing to google cal (Discussion #406)


What do you see in the execution log if you manually run startSync? What you're showing is a successful install, but that function doesn't do anything other than create triggers really (i.e. it doesn't process calendar events).

**Comment by Lonestarjeepin** [comment]
What error(s) are you getting?  Anything you can provide to help troubleshooting will be helpful.

**Comment by JohnGainor1** [comment]
Does this help?
Sorry I am just a hack 😉

Execution log
1:16:15 PM
Notice
Execution started
1:16:17 PM

Error
Bad host name: ***@***.***/8bb6338465a542d7a0d3589381dfcede15483958006230693011/calendar.ics

callWithBackoff
@ Helpers.gs:1078

fetchSourceCalendars
@ Helpers.gs:136

startSync
@ Code.gs:176


From: Lonestarjeepin ***@***.***>
Reply-To: derekantrican/GAS-ICS-Sync ***@***.***>
Date: Monday, January 22, 2024 at 1:32 PM
To: derekantrican/GAS-ICS-Sync ***@***.***>
Cc: John Gainor ***@***.***>, Author ***@***.***>
Subject: Re: [derekantrican/GAS-ICS-Sync] Outlook 365 v16.81 on Sonoma not syncing to google cal (Discussion #406)


What error(s) are you getting? Anything you can provide to help troubleshooting will be helpful.

**Reply by JohnGainor1** [reply]
Just to confirm bad host error above - I found an error in my ICS entry
ok so here are remaining errors

<img width="1244" alt="Copy_of_GAS-ICS-Sync_-_Project_Editor_-_Apps_Script_and_Sent_Mail_•_2___Rincon" src="https://github.com/derekantrican/GAS-ICS-Sync/assets/157310179/cd64d954-ac0e-4a1d-b628-ecd7e4c17967">

this is from RUN in startSync

**Reply by jonas0b1011001** [reply]
You have to remove the unused sample entries in `var sourceCalendars` aswell as the `,` at the end of your remaining entry.
https://github.com/derekantrican/GAS-ICS-Sync/blob/17cde1a768ad4c6d5ede0e7401e20606fff0be2a/Code.gs#L29-L30

**Comment by Lonestarjeepin** [comment]
The error for icsUrl2 is just a leftover example in the sourceCalendars array (row 29 in Code.gs).  Just delete any rows that you aren't using (icsURL2 and probably icsUrl3).  It looks like you got your main ics file to run successfully though.  Do you see all of the events in your Google Calendar now?

**Reply by JohnGainor1** [reply]
ahh got it
thanks for your help!
